src/modules/sync/components/sync-dialog.tsx

```diff
@@ -18,60 +19,101 @@ import { useDefaultSyncTree } from 'src/modules/sync/hooks/use-default-sync-tree
 import { useSyncDialogTexts } from 'src/modules/sync/hooks/use-sync-dialog-texts';
 import { getSiteEnvironment } from 'src/modules/sync/lib/environment-utils';
 import { useI18nLocale } from 'src/stores';
+import { useLatestRewindId, useRemoteFileTree } from 'src/stores/sync';
+import { TreeViewLoadingSkeleton } from './tree-view-loading-skeleton';
 import type { SyncSite } from 'src/hooks/use-fetch-wpcom-sites/types';
 
 type SyncDialogProps = {
 	type: 'push' | 'pull';
 	localSite: SiteDetails;
 	remoteSite: SyncSite;
-	onSubmit: ( syncData: TreeNode[] ) => void;
+	onPush: ( syncData: TreeNode[] ) => void;
+	onPull: ( syncData: TreeNode[], rewindId: string ) => void;
 	onRequestClose: () => void;
 };
 
 const useDynamicTreeState = (
 	type: 'push' | 'pull',
 	localSiteId: string,
+	remoteSiteId: number | undefined,
 	setTreeState: React.Dispatch< React.SetStateAction< TreeNode[] > >
 ) => {
 	const wpFolders = useMemo( () => [ ...GRANULAR_SYNC_FOLDERS ], [] );
 	const wpContent = useContentFolders( localSiteId, wpFolders );
+	const { rewindId, isLoading: isLoadingRewindId } = useLatestRewindId( remoteSiteId, {
+		skip: type === 'push',
+	} );
+	const { fetchChildren } = useRemoteFileTree();
 
+	// Pre-check all options when there's no rewind ID
```

Comment:
```suggestion
	// If the site was just created and if there is no rewind_id yet, then we pre-check all options to visually show the user that they can just do full sync
```

Reply:
I rephrased the comment a bit: https://github.com/Automattic/studio/pull/1596/commits/97a19ba209dfef02163fd18b654bae5dab3381e3